servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/ServiceTalkHttpAttributesGetter.java

```diff
@@ -118,14 +121,22 @@ public String getUrlFull(final HttpRequestMetaData request) {
             if (effectiveHostAndPort == null) {
                 return null;
             }
-            StringBuilder sb = new StringBuilder();
             String scheme = request.scheme();
-            sb.append(scheme == null ? "http" : scheme)
+            if (scheme == null) {
+                scheme = HTTP_SCHEME;
+            }
+            String requestTarget = request.requestTarget();
+            StringBuilder sb = new StringBuilder(
+                    scheme.length() + 3 +
+                    effectiveHostAndPort.hostName().length() +
+                    ((effectiveHostAndPort.port()) >= 0 ? 5 : 0) +
+                    requestTarget.length());
+            sb.append(scheme == null ? HTTP_SCHEME : scheme)
```

Comment:
Looks like `scheme` is always non-null here because of the check in lines 125-127